### PR TITLE
Adding Portuguese localization

### PR DESCRIPTION
**Pull request by milliet**
Files created and modified to add "pt" (Portuguese) to languages available.

### CODE REVIEW

**Comment by horus68** [comment]
Good effort, you still have many errors to correct in the language, or is this a Portuguese (Brasil) translation? 
If so the language code needs to be different to have different files for Portugal and Brasil !

Note: I've been waiting that my proposal to create a Crowdin or Transifex system is implemented!

**Comment by milliet** [comment]
Yes, this is a Brazilian Portuguese language. The translation is already
finished. We have built a working Mac version, but still couldn't make a
windows one.

Anyway, you're right about having a pt and a pt-br versions in the Lang
catalog. We'll do it, bit it seems we have to change all catalogs.

The European portuguese would need minor adjustments, but it is almost the
same.

Thanks

Pedro

Em sáb, 8 de ago de 2020 15:15, Horus68 <notifications@github.com> escreveu:

**Comment by horus68** [comment]
Thanks. If you can please duplicate the translations to have a pt-PT and a pt-BR version for this PR.
I will then update the pt-PT version later, when merged.
@danielweck requesting your yes on this!

File: src/resources/locales/pt.json [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+    },
+    "settings": {
+        "keyboard": {
+            "advancedMenu": "Menu",
+            "cancel": "Cancelar",
+            "edit": "Editar",
+            "editUserJson": "Editar arquvo de usuário...",
+            "hide": "Ocultar",
+            "keyboardShortcuts": "Atalhos de teclado",
+            "loadUserJson": "Carregar de arquivo de usuário",
```
Comment (horus68) [inline src/resources/locales/pt.json:322]: typo: change arquvo to arquivo

File: src/resources/locales/pt.json [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            "landmarks": "Pontos de referência",
+            "toc": "Índice"
+        },
+        "media-overlays": {
+            "activate": "Ativar Narração",
+            "captions": "Legendas",
+            "next": "Próxima narração",
+            "pause": "Pausar narração",
+            "play": "Tocar narração",
+            "previous": "Naração anterior",
```
Comment (horus68) [inline src/resources/locales/pt.json:240]: Typo: change "Naração" to "Narração"

File: src/resources/locales/pt.json [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            "addExtract": "Importar o excerto",
+            "goBuyBook": "Comprar este livro",
+            "goLoanBook": "Emprestar este livro",
+            "goSubBook": "Subscrever"
+        },
+        "network": {
+            "error": "Incapaz de acessar a origem OPDS",
+            "noInternet": "Sem Internet",
+            "noInternetMessage": "Tente: checar o cabo de rede, modem, router ou wifi",
+            "reject": "Desculpe, houve um erro inesperado tentanto processar a origem OPDS",
```
Comment (horus68) [inline src/resources/locales/pt.json:182]: Tyo: change "tentanto" to "tentando"

File: src/resources/locales/pt.json [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+                "holdPosition": "Sua posição de espera",
+                "holdTotal": "Reservado",
+                "numberOfItems": "Número de ítens",
+                "priveValue": "Preço",
+                "state": "Estado"
+            }
+        },
+        "publisher": "Editora",
+        "readBook": "Ler",
+        "released": "Pubolicado em",
```
Comment (horus68) [inline src/resources/locales/pt.json:94]: Typo: change "Pubolicado" to "Pubilicado"

**Comment by milliet** [comment]
Thank you @horus68! I'll make the corrections  you've pointed out and include pt-pt as you suggest.

**Comment by danielweck** [comment]
Thank you Pedro and "Horus68", much appreciated :)

**Comment by milliet** [comment]
Hi @horus68! I hope you're fine there. I've fixed the typos and created files for pt-pt.json and pt-br.json, pt-pt.md and pt-br.md (in src/resources/information), updated ptLang.json (in src/resources/locale-names) to include both versions for Portuguese (Portugal and Brasileiro). I didn't change /src/common/services/translator.ts because I didn't know what to do with it, but it seems it must be changed to allow four letter languages. 
Please, tell me if I need to do any other change.

Hi @danielweck. Nice to be here! Thanks for helping.

**Comment by horus68** [comment]
@milliet Thanks for your work.
Hope this PR gets merged soon. Then I can work to improve the pt-PT files.
@danielweck It would also require a dev to implement the four letter code languages. This should be a system wide config as there are many other languages requiring this!

File: src/common/services/translator.ts [hunk 5]
```diff
@@ -13,13 +13,15 @@ import * as esCatalog from "readium-desktop/resources/locales/es.json";
 import * as frCatalog from "readium-desktop/resources/locales/fr.json";
 import * as jaCatalog from "readium-desktop/resources/locales/ja.json";
 import * as nlCatalog from "readium-desktop/resources/locales/nl.json";
+import * as ptCatalog from "readium-desktop/resources/locales/pt.json";
```
Comment (danielweck) [inline src/common/services/translator.ts:16]: This should be:

```
import * as ptBrCatalog from "readium-desktop/resources/locales/pt-br.json";
import * as ptPtCatalog from "readium-desktop/resources/locales/pt-pt.json";
```

File: src/common/services/translator.ts [hunk 6]
```diff
@@ -95,6 +97,9 @@ i18nextInstance.init({
         ja: {
             translation: jaCatalog,
         },
+        pt: {
+            translation: ptCatalog,
+        },
```
Comment (danielweck) [inline src/common/services/translator.ts:102]: This should be:

```
        ptBr: {
            translation: ptBrCatalog,
        },
        ptPt: {
            translation: ptPtCatalog,
        },
```

File: src/common/services/translator.ts [hunk 7]
```diff
@@ -122,6 +127,7 @@ i18nextInstance.addResourceBundle("de", "translation", deLang, true);
 i18nextInstance.addResourceBundle("es", "translation", esLang, true);
 i18nextInstance.addResourceBundle("nl", "translation", nlLang, true);
 i18nextInstance.addResourceBundle("ja", "translation", jaLang, true);
+i18nextInstance.addResourceBundle("pt", "translation", ptLang, true);
```
Comment (danielweck) [inline src/common/services/translator.ts:130]: This should be:

```
i18nextInstance.addResourceBundle("pt-br", "translation", ptBrLang, true);
i18nextInstance.addResourceBundle("pt-pt", "translation", ptPtLang, true);
```

File: src/common/services/translator.ts [hunk 8]
```diff
@@ -140,6 +146,7 @@ export const AvailableLanguages = {
     es: "Español",
     nl: "Dutch",
     ja: "日本語",
+    pt: "Português",
```
Comment (danielweck) [inline src/common/services/translator.ts:149]: This should be:

```
"pt-br": "Português Brasileiro",
"pt-pt": "Português",
```

File: src/common/services/translator.ts (hunk 6 above)
Comment (danielweck) [reply src/common/services/translator.ts:102]: Sorry, wrong key syntax, I meant:

```
       pt-br: {
            translation: ptBrCatalog,
        },
        pt-pt: {
            translation: ptPtCatalog,
        },
```

**Comment by danielweck** [comment]
Thank you very much! I added comments to the PR source code.

**Comment by horus68** [comment]
@milliet Pedro... your turn now!

**Comment by milliet** [comment]
@danielweck, translator.ts Done! Tell me if you need any other action. Thanks.
@horus68 , it seems it is fine for now, but tell me about anything needed.

**Comment by horus68** [comment]
@milliet ptLang.jso should also be duplicated with the proper name code change

File: src/common/services/translator.ts [hunk 9]
```diff
@@ -13,13 +13,16 @@ import * as esCatalog from "readium-desktop/resources/locales/es.json";
...
+import * as ptBrCatalog from "readium-desktop/resources/locales/pt-br.json";
+import * as ptPtCatalog from "readium-desktop/resources/locales/pt-pt.json";
 
 import * as deLang from "readium-desktop/resources/locale-names/deLang.json";
 import * as enLang from "readium-desktop/resources/locale-names/enLang.json";
 import * as esLang from "readium-desktop/resources/locale-names/esLang.json";
 import * as frLang from "readium-desktop/resources/locale-names/frLang.json";
 import * as jaLang from "readium-desktop/resources/locale-names/jaLang.json";
 import * as nlLang from "readium-desktop/resources/locale-names/nlLang.json";
+import * as ptLang from "readium-desktop/resources/locale-names/ptLang.json";
```
Comment (horus68) [inline src/common/services/translator.ts:25]: The file ptLang.json needs to be duplicated for pt-pt and pt-br

**Review by horus68 — CHANGES_REQUESTED** [review]

File: src/common/services/translator.ts (hunk 9 above)
Comment (milliet) [reply src/common/services/translator.ts]: Changes made as suggested, 1 file created (pt-brLang.json) and one renamed (pt-ptLang.json). translator.ts changed to replace ptLang.json by both files. Thanks and waiting for any other change needed.

**Comment by milliet** [comment]
@horus68 and @danielweck , it seems it's OK now. I had to change all lang codes to quoted in translator.ts (it didn't pass Mac and Linux before that).

**Comment by danielweck** [comment]
@horus68 could you please “resolve conversation” (open comments) wherever appropriate in the discussion thread? Once confirmed, we will merge the PR.

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Review by horus68 — APPROVED** [review]

**Comment by horus68** [comment]
@danielweck I think I approved it all now (this github feature its not very intuitive)

**Pull request merged**